### PR TITLE
feat: add telemetry for office add-in treeview handlers

### PR DESCRIPTION
**Pull request by richard6094**
Add or adjust telemetry for office add-in treeview handlers.

### CODE REVIEW

File: packages/vscode-extension/src/handlers.ts [hunk 1]
```diff
@@ -375,9 +374,18 @@ export async function createNewProjectHandler(args?: any[]): Promise<Result<any,
...
   // show local debug button by default
+  const triggerFrom = getTriggerFromProperty(args);
   if (isValidOfficeAddInProject(projectPathUri.fsPath)) {
+    ExtTelemetry.sendTelemetryEvent(TelemetryEvent.CreateProjectStart, {
+      ...triggerFrom,
+      [TelemetryProperty.IsOfficeAddIn]: "true",
+    });
     await openOfficeDevFolder(projectPathUri, true, res.warnings, args);
   } else {
+    ExtTelemetry.sendTelemetryEvent(
```
Comment (hermanwenhe) [inline packages/vscode-extension/src/handlers.ts:385]: Better add [TelemetryProperty.IsOfficeAddIn]: "false" even when it's not Office Add-in.

File: packages/vscode-extension/src/telemetry/extTelemetryEvents.ts [hunk 2]
```diff
@@ -256,6 +256,15 @@ export enum TelemetryEvent {
   ShowScaffoldingWarningSummaryError = "show-scaffolding-warning-summary-error",
 
   FindSimilarIssues = "find-similar-issues",
+
+  //Office add-in related
+  validateAddInManifest = "validate-addin-manifest",
+  installAddInDependencies = "install-addin-dependencies",
+  stopAddInDebug = "stop-office-addin-debug",
+  generateAddInGUID = "generate-addin-guid",
+  editAddInManifest = "edit-addin-manifest",
```
Comment (hermanwenhe) [inline packages/vscode-extension/src/telemetry/extTelemetryEvents.ts:265]: editAddInManifest should be deleted

File: packages/vscode-extension/src/officeDevHandlers.ts [hunk 3]
```diff
@@ -93,13 +139,21 @@ export async function openScriptLabLink(args?: any[]): Promise<Result<boolean, F
 }
 
 export function validateOfficeAddInManifest(args?: any[]): Promise<Result<null, FxError>> {
+  ExtTelemetry.sendTelemetryEvent(
+    TelemetryEvent.validateAddInManifest,
+    getTriggerFromProperty(args)
```
Comment (hermanwenhe) [inline packages/vscode-extension/src/officeDevHandlers.ts:144]: Add IsOfficeAddIn property in office only event

File: packages/vscode-extension/src/officeDevHandlers.ts [hunk 4]
```diff
@@ -114,9 +168,15 @@ export async function popupOfficeAddInDependenciesMessage() {
...
     // Handle Yes button click
+    ExtTelemetry.sendTelemetryEvent(TelemetryEvent.popupOfficeAddInInstallMessage, {
+      [TelemetryProperty.Success]: true.toString(),
+    });
     await autoInstallDependencyHandler();
   } else if (result === "No") {
     // Handle No button click
+    ExtTelemetry.sendTelemetryEvent(TelemetryEvent.popupOfficeAddInInstallMessage, {
+      [TelemetryProperty.Success]: false.toString(),
+    });
```
[the remark this answers is not in the document]
Comment (richard6094) [reply packages/vscode-extension/src/officeDevHandlers.ts:179]: No. Shell we add some logic here to check if user have no operation on this message box?

**Pull request closed without merging**